mlc_llm/core.py

```diff
@@ -359,16 +359,16 @@ def mod_transform_before_build(
         mod
     )  # pylint: disable=not-callable
 
-    if "num_attention_heads" in config and "hidden_size" in config:
+    if config.num_attention_heads and config.hidden_size:
         max_seq_len = None
         if args.max_seq_len > 0:
             max_seq_len = args.max_seq_len
-        elif "max_sequence_length" in config:
-            max_seq_len = config["max_sequence_length"]
+        elif config.max_sequence_length:
+            max_seq_len = config.max_sequence_length
 
         if max_seq_len:
             mod = fuse_split_rotary_embedding(
-                mod, config["num_attention_heads"], config["hidden_size"], max_seq_len
+                mod, config.num_attention_heads, config.hidden_size, max_seq_len
```

Comment:
@MasterJH5574 @nverke Could you please confirm that this fixes the bug outlined in https://github.com/mlc-ai/mlc-llm/pull/806#issuecomment-1696766685?

Reply:
I can test this out tomorrow! Must have accidentally tested the wrong branch 🤦‍♂️

Reply:
Thank you!

Reply:
@sudeepag Sorry we keep making bugs with this issue. Can you make a separate PR for this change to merge this quickly?

Reply:
I think #826 fixes the issue. Could you try drop all changes here and rebase on to the latest main?

Reply:
Created https://github.com/mlc-ai/mlc-llm/pull/831 to add a test for this and resolve the error that is handled here.

Reply:
The hasattr is not enough because the objects are no longer dicts and therefore cannot be subscripted.